src/Package/Impl/Utilities/SessionUtilities.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.R.Host.Client;
+using Microsoft.R.Host.Client.Session;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.R.Package.Shell;
+
+namespace Microsoft.VisualStudio.R.Package.Utilities {
+    public static class SessionUtilities {
+        public static IRSession GetInteractiveSession() {
```

Comment:
We should not create extensions like this one, cause it couples `Microsoft.R.Components` object to the `VsAppShell.Current` instance. That is bad because:
- Any instance of the object from `Microsoft.R.Components` can live longer than `VsAppShell` and created before it. Calling to `VsAppShell.Current` before it is initialized in Package breaks pretty much everything. 
- `VsAppShell.Current` implicitly adds dependency to the IVs* to the code that doesn't really need it (we have dozens of commands in `Package` that can be moved to the `Microsoft.R.Components`).
- Any code that uses `VsAppShell.Current` requires isolated non-parallel run, lots of fakes and VsRHostScript even if the test has no UI involved.